src/odemis/driver/test/smaract_test.py

```diff
@@ -720,6 +720,185 @@ def test_parse_tsv_config(self):
+        f = self.dev.moveAbs(new_pos)
+        f.cancel()
+
+        difference = new_pos['z'] - self.dev.position.value['z']
+        self.assertNotEqual(round(difference, 5), 0)
+
+        # Test cancellation by stopping
+        self.dev.moveAbs({'z': SAFE_Z}).result()
+        new_pos = {'z': SAFE_Z + BIG_STEP}
+        f = self.dev.moveAbs(new_pos)
+        time.sleep(0.05)
+        self.dev.stop()
+
+        difference = new_pos['z'] - self.dev.position.value['z']
+        self.assertNotEqual(round(difference, 4), 0)
+
+    def test_move_rel(self):
+        # Test relative moves
+        self.dev.moveAbs({'z': SAFE_Z}).result()
+        old_pos = self.dev.position.value
+        shift = {'z': BIG_STEP}
+        self.dev.moveRel(shift).result()
+        new_pos = self.dev.position.value
+
+        testing.assert_pos_almost_equal(smaract.add_coord(old_pos, shift), new_pos, **COMP_ARGS)
+
+        # Test several relative moves and ensure they are queued up.
+        old_pos = self.dev.position.value
+        shift = {'z': -0.000001}
+        self.dev.moveRel(shift)
+        self.dev.moveRel(shift)
+        self.dev.moveRel(shift).result()
+
+        new_pos = smaract.add_coord(smaract.add_coord(smaract.add_coord(old_pos, shift), shift), shift)
+        testing.assert_pos_almost_equal(self.dev.position.value, new_pos, **COMP_ARGS)
+
+    def test_reference_cancel(self):
+        """Test canceling referencing"""
+        axes = set(self.dev.axes.keys())
+        # First, reference it
+        f = self.dev.reference(axes)
+        f.result()
+        for a, i in self.dev.referenced.value.items():
+            self.assertTrue(i)
+
+        f = self.dev.reference(axes)
+        time.sleep(0.1)
+        f.cancel()
+
+        for a, i in self.dev.referenced.value.items():
+            self.assertFalse(i)
+
+    def test_reference(self):
+        axes = set(self.dev.axes.keys())
+        f = self.dev.reference(axes)
+        f.result()
+
+        for a, i in self.dev.referenced.value.items():
+            self.assertTrue(i)
+
+        # Try again, after a move
+        shift = {'z': BIG_STEP}
+        self.dev.moveRel(shift).result()
+        pos_move = dict(self.dev.position.value)
+
+        f = self.dev.reference(axes)
+        f.result()
+        pos_refd = dict(self.dev.position.value)
+
+        for a, i in self.dev.referenced.value.items():
+            self.assertTrue(i)
+
+        # Check that at least the position changed
+        self.assertNotEqual(pos_move["z"], pos_refd["z"])
+
+    def test_reference_and_deactivate_move(self):
+        # Set a deactive position and check to be sure that the controller moves to this location
+        # after a reference move
+        de_pos = {'z': REF_Z}
+        self.dev.updateMetadata({model.MD_FAV_POS_DEACTIVE: de_pos})
+
+        f = self.dev.reference(set(self.dev.axes.keys()))
+        f.result()
+
+        testing.assert_pos_almost_equal(self.dev.position.value, de_pos, **COMP_ARGS)
```

Comment:
This whole test doesn't make sense. In the parameters `pos_deactive_after_ref=False`. So after referencing the actuator might be pretty much anywhere arbitrary, and that's expected. So you cannot expect here that it goes to "REF_Z". 
=> either set `pos_deactive_after_ref=True`, or remove this test.